### ISSUE TITLE
AzureResourceGroupDeployment@2 csmFile path not working if template artifact location contains space characters

### ISSUE THREAD

**Issue by petegrimsdale** [microsoft/azure-pipelines-tasks#11673]
**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**:  AzureResourceGroupDeployment@2

## Environment
- Server - Azure Pipelines
    
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: 
AZGlobal/Azure Global CAT Engineering/Pipelines/Releases/Benfica-Jmeter-release

- Agent - Hosted
        - If using Hosted agent, provide agent queue name:
Agent name: 'Azure Pipelines 2'
Agent machine name: 'fv-az645'
Current agent version: '2.159.2'

## Issue Description

Using the AzureResourceGroupDeployment@2 task and providing the file path link to the ARM template in the classic pipeline UI the task fails with error:
Error: Could not find any file matching the template file pattern

![image](https://user-images.githubusercontent.com/46575174/67961239-58fb4e00-fbf3-11e9-8e97-466e38d29f32.png)

The debug logs show the path to the template correctly but failing when "applying matching pattern" within findmatch code block

### Task logs

[tasklog_5.zip](https://github.com/microsoft/azure-pipelines-tasks/files/3794596/tasklog_5.zip)

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

### Error logs

[Insert error from the logs here for a quick overview]

**Comment by vinodkumar3** [comment]
For finding matching file patters AzureResourceGroupDeployment@2 task use [this](https://github.com/microsoft/azure-pipelines-task-lib/blob/master/node/docs/azure-pipelines-task-lib.md#taskfindMatch) library method. 
I'm able to repro the scenario and the way to get unblocked is to avoid using brackets ('[' or ']') in folder/file name.
To get more info over behavior of findMatch library method, feel free to open an issue in [azure-pipelines-task-lib](https://github.com/microsoft/azure-pipelines-task-lib) repo,

**Comment by vinodkumar3** [comment]
Closing this issue in favor of main tracking [issue](https://github.com/microsoft/azure-pipelines-task-lib/issues/585).

**Comment by bishal-pdMSFT** [comment]
Reactivating this issue till the time tracking issue is triaged as a bug

**Comment by jalajmsft** [comment]
Hey @petegrimsdale The fix is rolled out and should be available in all the regions by end of this week. Closing this issue. Feel free to re-open or create a new issue for further queries.
